### DISCUSSION TITLE
axios http interceptors in nuxt3

### DISCUSSION THREAD

**Discussion by Matman94** [nuxt/nuxt#16565] [General]
Please provide me any solution or documentation link (I can't find) for how to make a plugin (or something like that) where can I intercept request and response hooks on http methods. In nuxt 2 it was a very simply, I made plugin and registered it in nuxt.config.js and that's all. 
I nuxt3 I can't so easly do it...

**Comment by Diizzayy** [comment]
@Matman94 Nuxt 3 uses [ohmyfetch](https://github.com/unjs/ohmyfetch) to make requests by default, You can create a custom fetch [composable](https://v3.nuxtjs.org/guide/directory-structure/composables) and utilize the [ohmyfetch interceptors ](https://github.com/unjs/ohmyfetch#%EF%B8%8F-interceptors)

`nuxt-app/composables/useCustomFetch.ts`
```ts
export const useCustomFetch = (url: string, options?: FetchOptions) => {
  return useFetch(url, {
    ...options,
    async onResponse({ request, response, options }) {
      console.log('[fetch response]')
    },
    async onResponseError({ request, response, options }) {
      console.log('[fetch response error]')
    },

    async onRequest({ request, options }) {
      console.log('[fetch request]')
    },
    async onRequestError({ request, options, error }) {
      console.log('[fetch request error]')
    }
  })
}
```

## Usage

```vue
<script lang="ts" setup>
const { data } = await useCustomFetch('/api/demo')
</script>
```

**Reply by Matman94** [reply]
Thank you for your answer. I need something more general like on every request or every response - decorator. I would like avoid composable in this case because if someone from my team forgets about it and use just fetch instead customized composable then sensitive and important code wrapped on axios/fetch/(other http client) hooks will not execute.
So, do you know how make this more general? I mean one file with onRequest/Response/Error handlers without import any additional files like composable or other files to component.
